plugin/pkg/scheduler/algorithm/predicates/predicates_test.go

```diff
@@ -346,15 +355,15 @@ func TestPodFitsHost(t *testing.T) {
 	for _, test := range tests {
 		nodeInfo := schedulercache.NewNodeInfo()
 		nodeInfo.SetNode(test.node)
-		result, err := PodFitsHost(test.pod, PredicateMetadata(test.pod, nil), nodeInfo)
-		if !reflect.DeepEqual(err, ErrPodNotMatchHostName) && err != nil {
-			t.Errorf("unexpected error: %v", err)
+		fits, reasons, err := PodFitsHost(test.pod, PredicateMetadata(test.pod, nil), nodeInfo)
+		if err != nil {
+			t.Errorf("%s: unexpected error: %v", test.test, err)
 		}
-		if result == false && !reflect.DeepEqual(err, ErrPodNotMatchHostName) {
-			t.Errorf("unexpected error: %v", err)
+		if !fits && !reflect.DeepEqual(reasons, []algorithm.PredicateFailureReason{ErrPodNotMatchHostName}) {
```

Comment:
instead of creating this slice in every loop, can you please create it once before the loop, i.e.:
```
expectedReasons := []algorithm.PredicateFailureReason{ErrPodNotMatchHostName}
```
and use it here.

Same for all others below.